### PR TITLE
feat(theme): overlay system (Modal/Drawer/Portal/FocusTrap) and control focus a11y

### LINKED ISSUE #181: [Task] Introduce a zOrder stacking scale and refactor AppBar to use it
### Task Description ✏️

Add a centralized **`zOrder`** stacking scale to the theme so layered UI (sticky header, drawers, modals) shares one ordered source of truth instead of scattered magic numbers. Today the header's stacking value is a hardcoded `zIndex={50}` in `Header.tsx`; there is no token governing what sits above what.

Introduce `zOrder` as a structural token on the `Theme` interface and the shared `baseTheme` in `apps/web/src/theme/themes.ts` (identical across light/dark), then refactor **`AppBar`** to default its own z-index from `theme.zOrder.appBar`, and remove the literal `zIndex={50}` from `Header.tsx`. This establishes the layering needed by upcoming overlay components (Modal, Drawer).

Parent epic: #40.

### Acceptance Criteria ✅

- `zOrder` added to the `Theme` interface and to `baseTheme` in `themes.ts`, with ascending role-named layers: `appBar` → `drawer` → `modal` (suggested values `appBar: 1100`, `drawer: 1200`, `modal: 1300`; values comfortably above existing content/`CookieNotice` z-indexes).
- `AppBar` applies `zIndex: theme.zOrder.appBar` by default, while still allowing an explicit `zIndex` override prop.
- `Header.tsx` no longer passes the literal `zIndex={50}`; the header still renders fixed and above page content (verified by test).
- No raw/hardcoded z-index numbers remain in `AppBar`/`Header`.
- Unit tests cover the new token presence and the AppBar default/override behavior; **100% coverage** on all touched files.

### Dependencies 🔗

- None. Unblocks the Modal/Drawer overlay work (separate tasks under #40).

### Definition of Ready (DoR) 📋

- Token name (`zOrder`), layer keys, and ordering agreed.
- `design-system.md` token conventions reviewed.

### Definition of Done (DoD) 🏁

- Code implemented and reviewed.
- `pnpm lint` clean (max-warnings 0), `pnpm test:coverage` at 100% on touched files, `pnpm build` passes.
- `AppBar` `README.md`/stories updated if the public prop surface changes.
- No coverage-ignore pragmas; imports use the `src/...` path alias.

### Additional Notes or Resources 📎

- Files: `apps/web/src/theme/themes.ts` (interface + `baseTheme`), `apps/web/src/theme/AppBar/AppBar.tsx` (+ test/README/stories), `apps/web/src/common/Header/Header.tsx` (+ test).
- Backdrop scrims stay relative to their modal root (`zIndex: -1`) and intentionally get **no** `zOrder` token.
- Reference: `src/theme/design-system.md`.

### PR DESCRIPTION
**Pull request by soroushm**
## Summary

This branch delivers two related bodies of work.

### 1. Overlay / modal system

- **zOrder stacking scale** + default AppBar z-index, so overlays sit above the sticky header. (#181)
- **Modal manager + `useModal` hook** for coordinating layered overlays. (#182)
- **`Portal`, `FocusTrap`, `Backdrop`, `Modal`** primitives. (#183)
- **`Drawer`** component built on `Modal`. (#184)
- **Mobile navigation `Drawer`** wired into the `Header`. (#185)

### 2. Focus & keyboard accessibility for interactive controls (#80)

- **`Switch`** — `Enter` now toggles the control (`Space` already toggled natively via the underlying checkbox), matching the ARIA switch pattern. Focus-ring color resolves from `theme.palette.primary.main` rather than `currentColor`.
- **`Checkbox` / `Radio`** — the icon wrapper's inline SVG was inflating the wrapper height via its line box (wrapper measured `20 × 23.33` instead of square), so the focus outline rendered as a tall rectangle around a square/circular control. Setting the wrapper to `line-height: 0` keeps it square and the outline evenly spaced. Ring color also resolves from `palette.primary.main`.
- **`Link`** — added a `:focus-visible` ring (`2px palette.primary.main`, `4px` offset, rounded); it previously had none. The `Header` logo was a raw `<a>` and now uses the `Link` component so it inherits the themed ring (`NavLink` already wraps `Link`).
- **`Button` / `TextInput`** — keyboard focus ring resolves from `palette.primary.main`.

## Verification

- `pnpm lint` — clean (zero warnings)
- `pnpm test:silent` — 206 files / 1846 tests passing
- 100% coverage on all touched files

## Closes

#181, #182, #183, #184, #185, #80

## Summary by CodeRabbit

* **New Features**
  * Added responsive navigation with a mobile menu in the header.
  * Introduced new overlay components and utilities for modals, drawers, backdrops, portals, and focus handling.
  * Added a new menu icon and expanded Storybook examples for several UI components.

* **Bug Fixes**
  * Improved layering, focus visibility, keyboard support, and scroll behavior across interactive components.
  * Updated modal and drawer behavior for better accessibility and mobile interactions.

* **Documentation**
  * Added and refreshed component docs for the new overlay and focus features.

### CODE REVIEW

File: apps/web/src/theme/Modal/utils/getScrollbarSize.test.ts [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+  it('returns the difference between innerWidth and clientWidth', () => {
+    expect(getScrollbarSize(makeWindow(1024, 1009))).toBe(15)
+  })
+
+  it('returns 0 when there is no scrollbar', () => {
+    expect(getScrollbarSize(makeWindow(1024, 1024))).toBe(0)
+  })
+
+  it('never returns a negative value', () => {
+    expect(getScrollbarSize(makeWindow(1000, 1024))).toBe(24)
```
[the remark this answers is not in the document]
Comment (soroushm) [reply apps/web/src/theme/Modal/utils/getScrollbarSize.test.ts:20]: Skipping — this is a false positive. `getScrollbarSize` uses `Math.abs(win.innerWidth - documentWidth)`, not a `Math.max(0, …)` clamp, so for `makeWindow(1000, 1024)` the result is `abs(-24) = 24`. The `toBe(24)` assertion is correct, and `Math.abs` already guarantees the "never negative" property the test name describes. Changing the expectation to `0` would require altering the implementation, which is out of scope for this PR.

File: apps/web/src/utils/ownerWindow.ts [hunk 2]
```diff
@@ -0,0 +1,10 @@
+import { ownerDocument } from 'src/theme/Modal/utils/ownerDocument'
+
+/**
+ * Returns the `Window` that owns the given node, falling back to the global
+ * `window`. Used to read viewport metrics for scroll-lock calculations.
+ */
+export function ownerWindow(node: Node | null | undefined): Window {
+  const doc = ownerDocument(node)
+  return doc.defaultView ?? window
```
[the remark this answers is not in the document]
Comment (soroushm) [reply apps/web/src/utils/ownerWindow.ts:9]: Skipping the `typeof window` guard by design. `ownerWindow` and `ownerDocument` are only ever invoked from the browser-only scroll-lock effects, and `ownerDocument` already dereferences the `document` global the same way — so the chain is intentionally browser-only. Adding a `typeof window` guard would force a `Window | undefined` return type that breaks every caller (the function contract is a non-optional `Window`). Leaving the direct reference.

Note: both helpers have now been promoted out of `theme/Modal/utils/` to `src/utils/ownerWindow.ts` and `src/utils/ownerDocument.ts`, since they're generic.

File: apps/web/src/common/Header/Header.tsx [hunk 3]
```diff
@@ -1,10 +1,17 @@
+import { useState } from 'react'
```
[the remark this answers is not in the document]
Comment (soroushm) [reply apps/web/src/common/Header/Header.tsx:1]: Skipping this one intentionally. The drawer here is the mobile navigation menu, and the scenario — a viewport crossing the `768px` breakpoint *while the menu is open* — is really a desktop-resize / devtools case rather than something that happens in normal mobile use, where the viewport width is effectively fixed.

It also never becomes genuinely stuck: the drawer is dismissable by backdrop click and Escape (inherited from `Modal`), and selecting any nav item closes it (the panel's `onClick` resets `isMenuOpen`). Given that, wiring a `matchMedia` listener — plus a jsdom `matchMedia` mock to keep the Header tests green — adds weight that isn't justified by the edge case. Leaving as-is.

File: apps/web/src/theme/Modal/ModalManager.ts [hunk 4]
```diff
@@ -0,0 +1,194 @@
...
+        modal.mount,
+        modal.modalRef,
+        containerState.hiddenSiblings,
+        false
+      )
+      this.containers.splice(this.containers.indexOf(containerState), 1)
+    } else {
+      // Reveal the next modal down the stack to assistive tech.
+      const nextTop = containerState.modals[containerState.modals.length - 1]
+      ariaHidden(nextTop.modalRef, false)
```
[the remark this answers is not in the document]
Comment (soroushm) [reply apps/web/src/theme/Modal/ModalManager.ts:195]: Adopted in part, with a correction. In this codebase `Modal.tsx` never wires `useModal`'s `portalRef` to `<Portal>`, so `mountNodeRef` is never set and `ManagedModal.mount` is **always null** — the modal root (`modalRef`) is portaled *directly* into the container as a sibling, not nested inside a `mount` node. Applying `ariaHidden(modal.mount, …)` / `ariaHidden(nextTop.mount, …)` therefore dereferences null → `null.setAttribute` (reproduced as a crash in the nested Modal story).

What I did instead in `b3f7bc8`:

```ts
} else {
  const nextTop = containerState.modals.at(-1)!
  ariaHidden(modal.modalRef, ariaHiddenState) // re-hide the removed (possibly shouldKeepMounted) modal
  ariaHidden(nextTop.modalRef, false)          // reveal the modal below
}
```

This keeps the valid half of your finding (the removed modal no longer leaks to the AT) and reveals the next modal. Because the roots are portaled as direct siblings of the container, `ariaHiddenSiblings` already hid the lower root when the upper opened, so un-hiding that sibling root is sufficient — there's no separate `mount` ancestor to restore. Added a `Modal`-level regression test that opens two stacked modals and unmounts the top one to guard the crash.

(Wiring `portalRef` so `mount` becomes meaningful is a separate latent cleanup, tracked outside this PR.)

File: apps/web/src/theme/Modal/ModalManager.test.ts [hunk 5]
```diff
@@ -0,0 +1,206 @@
...
+  // Reset any viewport overrides applied by scroll-lock tests.
+  Object.defineProperty(window, 'innerWidth', { value: 1024, configurable: true })
+})
+
+/** Creates a managed modal whose mount node lives inside `parent`. */
+const makeModal = (parent: HTMLElement = container): ManagedModal => {
+  const mount = document.createElement('div')
+  const modalRef = document.createElement('div')
+  parent.append(mount)
+  return { mount, modalRef }
```
[the remark this answers is not in the document]
Comment (soroushm) [reply apps/web/src/theme/Modal/ModalManager.test.ts:23]: Keeping the fixture's `modalRef` detached from `mount` on purpose. It mirrors the **actual** DOM this codebase produces: `Modal.tsx` doesn't wire `useModal`'s `portalRef`, so `mount` is never populated and the modal root is portaled *directly* into the container (a sibling of the other modals' roots), not nested inside a `mount` node. Appending `modalRef` under `mount` would model a portal shape we don't actually render, and the manager only toggles `modalRef` (never `mount`) — see the reply on the `remove()` thread.

The nested reveal/hide path is covered by the new `re-hides the removed modal and reveals the next when a modal remains` test here, plus a `Modal`-level regression test that unmounts the top of a two-modal stack.
